improver/utilities/solar.py

```diff
@@ -224,14 +227,29 @@ def _create_daynight_mask(self, cube):
                 will be ignored although they might appear as attributes
                 on the cube as it is extracted from the first slice.
         """
-        daynight_mask = next(
-            cube.slices(
-                [cube.coord("time"), cube.coord(axis="y"), cube.coord(axis="x")]
-            )
-        ).copy()
-        daynight_mask.rename("day_night_mask")
-        daynight_mask.units = unit.Unit("1")
-        daynight_mask.data = np.ones(daynight_mask.data.shape, dtype="int") * self.night
+        slice_coords = [cube.coord(axis="y"), cube.coord(axis="x")]
+        if cube.coord("time") in cube.coords(dim_coords=True):
+            slice_coords.insert(0, cube.coord("time"))
+
+        template = next(cube.slices(slice_coords))
+        demoted_coords = [
+            crd
+            for crd in cube.coords(dim_coords=True)
+            if crd not in template.coords(dim_coords=True)
+        ]
+        [template.remove_coord(crd) for crd in demoted_coords]
```

Comment:
Codacy found an issue: [Expression "[template.remove_coord(crd) for crd in demoted_coords]" is assigned to nothing](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5829826)